react/src/views/BoxCreate/BoxCreateView.tsx

```diff
@@ -65,25 +69,84 @@ export const ALL_PRODUCTS_AND_LOCATIONS_FOR_BASE_QUERY = gql`
 `;
 
 function BoxCreateView() {
+  // Basics
+  const navigate = useNavigate();
+
+  // variables in URL
   const baseId = useParams<{ baseId: string }>().baseId!;
-  const [searchParams] = useSearchParams();
-  const qrCode = searchParams.get("qrCode") as string | undefined;
+  const qrCode = useParams<{ qrCode: string }>().qrCode!;
+
+  // Query the QR-Code
+  const qrCodeExists = useQuery<CheckIfQrExistsInDbQuery, CheckIfQrExistsInDbQueryVariables>(
+    CHECK_IF_QR_EXISTS_IN_DB,
+    {
+      variables: { qrCode },
+    },
+  );
 
-  const { loading, error, data } = useQuery<
+  // Query Data for the Form
+  const allFormOptions = useQuery<
     AllProductsAndLocationsForBaseQuery,
     AllProductsAndLocationsForBaseQueryVariables
   >(ALL_PRODUCTS_AND_LOCATIONS_FOR_BASE_QUERY, {
     variables: {
       baseId,
     },
   });
-  const navigate = useNavigate();
 
+  // Mutation after form submission
   const [createBoxMutation, createBoxMutationState] = useMutation<
     CreateBoxMutation,
     CreateBoxMutationVariables
   >(CREATE_BOX_MUTATION);
 
+  // Check the QR Code
+  useEffect(() => {
+    if (qrCodeExists.data?.qrExists === false) {
+      notificationVar({
```

Comment:
Note for later that this can be refactored with the enums that Vahid introduced in his PR.

Reply:
In general, we should refactor this business logic and put it into a hook. @pylipp @vahidbazzaz 
This is just a quick fix.